runtime/js/effect.js

```diff
@@ -215,3 +254,62 @@ function caml_ml_condition_signal(t) {
 function jsoo_effect_not_supported() {
   caml_failwith("Effect handlers are not supported");
 }
+
+//Provides: caml_resume
+//Requires:caml_stack_depth, caml_call_gen_cps, caml_exn_stack, caml_fiber_stack, caml_wrap_exception, caml_uncaught_effect_handler, caml_resume_stack
+//If: effects
+//If: doubletranslate
+function caml_resume(f, arg, stack) {
+  var saved_stack_depth = caml_stack_depth;
+  var saved_exn_stack = caml_exn_stack;
+  var saved_fiber_stack = caml_fiber_stack;
+  try {
+    caml_exn_stack = 0;
+    caml_fiber_stack = {
+      h: [0, 0, 0, { cps: caml_uncaught_effect_handler }],
+      r: { k: 0, x: 0, e: 0 },
+    };
+    var k = caml_resume_stack(stack, (x) => x);
+    /* Note: f is not an ordinary function but a (direct-style, CPS) closure pair */
+    var res = { joo_tramp: f, joo_args: [arg, k], joo_direct: 0 };
+    do {
+      caml_stack_depth = 40;
```

Comment:
can we have a comment explaining this ?

Reply:
This value is the same as in `caml_callback`. I guess it has been chosen because empirically, it avoids trampolining too often but still avoids overflow. I added a comment saying this.